app/models/katello/glue/candlepin/pool.rb

```diff
@@ -165,8 +165,13 @@ def create_product_associations
       def import_hosts
         uuids = Resources::Candlepin::Pool.consumer_uuids(self.cp_id)
 
-        sub_facet_ids_from_cp = Katello::Host::SubscriptionFacet.where(:uuid => uuids).select(:id).pluck(:id)
+        sub_facets_from_cp = Katello::Host::SubscriptionFacet.where(:uuid => uuids).select([:id, :host_id])
+        host_ids_from_cp = sub_facets_from_cp.pluck(:host_id)
+        sub_facet_ids_from_cp = sub_facets_from_cp.pluck(:id)
```

Comment:
This would trigger two sql queries, one to get the id and the other to get the host_id. instead you can get both in one query and transform in memory (it shouldn't be more than a few KB even for a large setup):
```ruby
sub_facet_ids_from_cp, host_ids_from_cp = Katello::Host::SubscriptionFacet.where(:uuid => uuids).pluck([:id, :host_id]).transpose